frontend/src/components/Pages/InspectionPage/InspectionSection.tsx

```diff
@@ -109,87 +230,138 @@ export function InspectionSection({ refreshInterval }: RefreshProps) {
-        }
-        setOngoingMissions(newOngoingMissions)
+        openDialog()
+        const sortedInspections = inspections.sort(compareInspections)
+        setSelectedMissions(sortedInspections.map((i) => i.missionDefinition))
     }
 
     useEffect(() => {
         const id = setInterval(() => {
-            updateOngoingMissionsMap(deckMissions)
+            updateScheduledMissionsMap(deckMissions)
         }, refreshInterval)
         return () => clearInterval(id)
     }, [deckMissions])
 
     const getCardColor = (deckId: string) => {
         const inspections = deckMissions[deckId].inspections
         if (inspections.length === 0) return 'gray'
-        const isScheduled = (m: CondensedMissionDefinition) =>
-            Object.keys(ongoingMissions).includes(m.id) && ongoingMissions[m.id]
-        const sortedInspections = inspections.filter((i) => !isScheduled(i.missionDefinition)).sort(compareInspections)
+        const sortedInspections = inspections.sort(compareInspections)
+
         if (sortedInspections.length === 0) return 'green'
 
         const nextInspection = sortedInspections[0]
-        if (!nextInspection.deadline) return 'green'
-
-        const deadlineDays = getDeadlineInDays(nextInspection.deadline)
-        switch (true) {
-            case deadlineDays <= 1:
-                return 'red'
-            case deadlineDays > 1 && deadlineDays <= 14:
-                return 'yellow'
-            case deadlineDays > 7:
-                return 'green'
+
+        if (!nextInspection.deadline) {
+            if (!nextInspection.missionDefinition.inspectionFrequency) return 'gray'
+            else return 'red'
         }
+
+        return getDeadlineInspection(nextInspection.deadline)
     }
 
+    useEffect(() => {
+        const newAreas: DeckAreas = {}
+
+        Object.keys(deckMissions).map((deckId) => {
+            let string: string = ''
+            let areaNames: String[] = []
+            BackendAPICaller.getAreadByDeckID(deckMissions[deckId].deck.id).then((areas) => {
+                areaNames = []
+                string = ''
+                if (areas != null) {
+                    areas.forEach((area) => {
+                        if (!areaNames.includes(area.areaName))
+                            areaNames = areaNames.concat(area.areaName.toLocaleUpperCase())
+                    })
+                    areaNames = areaNames.sort()
+                    console.log(areaNames)
+                    areaNames.forEach((areaName) => {
+                        console.log(areaName)
+                        if (areaNames[areaNames.length - 1] == areaName) string = string + ' ' + areaName
+                        else string = string + ' ' + areaName + ' |'
+                    })
+
+                    newAreas[deckMissions[deckId].deck.id] = {
+                        areaString: string,
+                    }
+                    setAreas(newAreas)
+                }
+            })
+        })
```

Comment:
```suggestion
        Object.keys(deckMissions).map((deckId) => {
            BackendAPICaller.getAreadByDeckID(deckMissions[deckId].deck.id).then((areas) => {
                const formattedAreaNames = areas
                    .map((area) => {
                        return area.areaName.toLocaleUpperCase()
                    })
                    .sort()
                    .join(' | ')

                newAreas[deckMissions[deckId].deck.id] = {
                    areaString: formattedAreaNames,
                }
            })
        })
        setAreas(newAreas)
```

It became quite a few overlapping comments here, so here is a complete suggestion for the entire part :)